engine/src/main/java/graphql/nadel/engine/execution/StrategyUtil.java

```diff
@@ -159,18 +159,28 @@ public static RootExecutionResultNode mergeTrees(List<RootExecutionResultNode> r
     private static ExecutionResultNode combineResultNodes(List<ExecutionResultNode> resultNodes) {
         List<ExecutionResultNode> children = new ArrayList<>();
         Map<String, Object> completedValueMap = new LinkedHashMap<>();
+        // TODO: feature flag check, perhaps?
+        List<ExecutionResultNode> nonNullNodes = resultNodes
```

Comment:
not 100 percent on this but I think we are trying to not use `stream` in nadel and use lightweight functions from `nadel/util/FpKit` where possible. I think Andi and Brad did some benchmark and found that there is non negligible impact when using java's `Streams`. This `stream.filter.collect` can be substituted with `nadel/util/FpKit.filter`